src/app/components/menus/categories.menu/categories.menu.component.ts

```diff
@@ -1,29 +1,47 @@
-import { Component, OnInit, Output, EventEmitter, Input } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { MenuController } from '@ionic/angular';
+import { Store, select } from '@ngrx/store';
 
-import { PostDataService } from '@data/post.data.service';
-import { ICategory } from '@interfaces/category';
-import { IPost } from '@interfaces/post';
+import { ICategory } from '@albert/interfaces/ICategory';
+import { IPost } from '@albert/interfaces/IPost';
+import { IPostFilter } from '@albert/interfaces/IPostFilter';
+import { PostDataService } from '@albert/data/post.data.service';
+import { setFilterCategory } from '@albert/store/actions';
 
 @Component({
   selector: 'app-categories-menu',
   templateUrl: './categories.menu.component.html',
   styleUrls: ['./categories.menu.component.scss'],
 })
-export class CategoriesMenuComponent implements OnInit {
-  @Input() posts: IPost[] = [];
-  @Output() setCategoryEvent = new EventEmitter();
-  
+export class CategoryMenuComponent implements OnInit {
+  posts: IPost[] = [];
   categories: ICategory[] = [];
-  currentCategoryId?: number | null;
+  currentCategoryId: number | undefined;
```

Comment:
Generally we put a question mark on the property instead to denote it's optional.

```suggestion
  currentCategoryId?: number;
```

However, there are nuances to this. Please check this [TypeScript Playground](https://www.typescriptlang.org/play/#code/PTAEBcCcE9QCwPYDcCmkCWA7A5qBmI4VQADABwENIKBbEvAM1IYQXoswBNSAjKkgFAhCKWDwTg4oAO7oANnNABnRNNJKoWXAB9QAVy4oGWFJ0HCee8KE4IIFANbFME4pJRK30u1WKd0DAxoKJjWWCLokPpKFNgoAgwGAMbg6PigLAgAFJTUNABcypo4oLoGnEYmnACUoADeAqCgwrbKCDQokloCAL4CCcmp6XyQOVS0APyFGhg4tQ1NLXZK7Z1w3X1CYAByEuhJxESQKAA0ylgHIqBkkAhkaOCw6OCeckzoSqAu1ndDmBSKLJJDigHjEABESgc6DI9044OqW1A0AQelAvlAACUAKIARQAqgBJHEAEVAABUAPKgAAK+PJoAAgqAAGqMgAy+OxCVYWWqAG5+sIONxJBRrElUXJuGD9IZjJhTDzsuVKoqaoKkR0ONJ1nJTqAaFRoSV3KRcrR6LJJOiCL80v9FHc0OKEJAkd5MAByazHACOekixBRaIYtxo1ysqRKFFASCo6AoPH1AhGfMFQA)